source/codegen/metadata/nirfsg/functions.py

```diff
@@ -0,0 +1,98 @@
+functions = {
+    "Init": {
+        "init_method": True,
+        "codegen_method": "public",
+        "cname": "niRFSG_init",
+        "parameters": [
+            {
+                "name": "resourceName",
+                "direction": "in",
+                "type": "ViRsrc"
+            },
+            {
+                "name": "idQuery",
+                "direction": "in",
+                "type": "ViBoolean"
+            },
+            {
+                "name": "reset",
+                "direction": "in",
+                "type": "ViBoolean"
+            },
+            {
+                "name": "vi",
+                "direction": "out",
+                "type": "ViSession",
+                "xxgrpc_type": "nidevice_grpc.Session"
+            }
+        ],
+        "returns": "ViStatus"
+    },
+    "Close": {
+        "cname": "niRFSG_close",
+        "parameters": [
+            {
+                "direction": "in",
+                "name": "vi",
+                "type": "ViSession"
+            }
+        ],
+        "returns": "ViStatus",
+    },
+    "InitWithOptions": {
+        "init_method": True,
+        "parameters": [
+            {
+                "direction": "in",
+                "name": "resourceName",
+                "type": "ViRsrc"
+            },
+            {
+                "direction": "in",
+                "name": "idQuery",
+                "type": "ViBoolean",
+            },
+            {
+                "direction": "in",
+                "name": "resetDevice",
+                "type": "ViBoolean"
+            },
+            {
+                "direction": "in",
+                "name": "optionString",
+                "type": "ViConstString",
+            },
+            {
+                "direction": "out",
+                "name": "vi",
+                "type": "ViSession"
+            }
+        ],
+        "returns": "ViStatus",
+    },
+    'ErrorMessage': {
+        'cname': 'niRFSG_error_message',
+        'parameters': [
+            {
+                'name': 'vi',
+                'direction': 'in',
+                'type': 'ViSession'
+            },
+            {
+                'name': 'errorCode',
+                'direction': 'in',
+                'type': 'ViStatus'
+            },
+            {
+                'name': 'errorMessage',
+                'direction': 'out',
+                'type': 'ViChar[]',
+                'size': {
+                    'mechanism': 'fixed',
+                    'value': 256
```

Comment:
The examples use `niRFSG_GetError`. In the RFSA header, this `error_message` variant is in a section for "Required VXIplug&play Functions". And, there are several idiosyncracies to those methods. Including not having a size mechanism. And I don't think any of them are in CVI. I have user stories to audit these as part of making sure we exclude everything that should be excluded. Or ... choose a legitimate fixed max size.

TL;DR: I think we're going to remove this method. Try using GetError instead,

Reply:
Ah, OK. Size issue aside, I like `error_message` better because it takes in an error code, instead of just returning the last one the session hit. But I switched this to `GetError` and updated the tests - thanks!

Reply:
Oops. I missed that. Yeah, I like that too. We'll figure it out.